src/nodemetadata.cpp

```diff
@@ -133,15 +140,19 @@ void NodeMetadataList::deSerialize(std::istream &is, IItemDefManager *item_def_m
 	u16 count = readU16(is);
 
 	for (u16 i=0; i < count; i++) {
-		u16 p16 = readU16(is);
-
 		v3s16 p;
-		p.Z = p16 / MAP_BLOCKSIZE / MAP_BLOCKSIZE;
-		p16 &= MAP_BLOCKSIZE * MAP_BLOCKSIZE - 1;
-		p.Y = p16 / MAP_BLOCKSIZE;
-		p16 &= MAP_BLOCKSIZE - 1;
-		p.X = p16;
-
+		if (absolute_pos) {
+			p.X = readS16(is);
+			p.Y = readS16(is);
+			p.Z = readS16(is);
+		} else {
+			u16 p16 = readU16(is);
+			p.X = p16 & (MAP_BLOCKSIZE - 1);
+			p16 /= MAP_BLOCKSIZE;
+			p.Y = p16 & (MAP_BLOCKSIZE - 1);
+			p16 /= MAP_BLOCKSIZE;
```

Comment:
@sfan5 Indeed, there is a missing masking in the last line (`p.Z = p16;`) but you can confirm that it's working by yourself: https://hastebin.com/ilekoqotil.cpp

@numberZero Yes, it may look a bit more confusing with masking. An AND operation is way faster than a modulo (unless it's optimized in the compiler).